flink/src/main/java/org/apache/iceberg/flink/sink/FlinkManifestUtil.java

```diff
@@ -63,4 +66,53 @@ static ManifestOutputFileFactory createOutputFileFactory(Table table, String fli
     TableOperations ops = ((HasTableOperations) table).operations();
     return new ManifestOutputFileFactory(ops, table.io(), table.properties(), flinkJobId, subTaskId, attemptNumber);
   }
+
+  static DeltaManifests writeCompletedFiles(WriteResult result,
```

Comment:
just for my own education, `referencedDataFiles` from `WriteResult` doesn't seem to be used (except for unit test). What is it for? do we need to serialize it too?

Reply:
We should serialize it and add it to the commit. This is the set of files that is referenced by any positional delete, which identifies deleted rows by file and row position. The commit will validate that all of the files still exist in the table.

This isn't strictly needed for this use case because we know that the position deletes only refer to files that are created in this commit. Since the files are being added in the commit, it isn't possible for some other process to delete some of them from metadata. But it is still good to configure the commit properly in case this gets reused later.

Reply:
Thanks for the explanation, @rdblue .   I think it's correct to validate the data files in `RowDelta#commit`.  Will provide an extra unit test to address it.

Reply:
This [unit test](https://github.com/apache/iceberg/pull/1939/files#diff-2bbb61f336190e507c0cb57fc6d01f86b081cf150b460086b89b035d3b4917c8R658) addressed the data files validation issue